packages/components/src/panel/body.js

```diff
@@ -2,94 +2,139 @@
  * External dependencies
  */
 import classnames from 'classnames';
+import { noop } from 'lodash';
+import {
+	useDisclosureState,
+	Disclosure,
+	DisclosureContent,
+} from 'reakit/Disclosure';
 
 /**
  * WordPress dependencies
  */
-import { Component, forwardRef } from '@wordpress/element';
+import { useReducedMotion } from '@wordpress/compose';
+import { forwardRef, useRef } from '@wordpress/element';
 import { chevronUp, chevronDown } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import Button from '../button';
 import Icon from '../icon';
+import { useCombinedRefs, useUpdateEffect } from '../utils';
 
-export class PanelBody extends Component {
-	constructor( props ) {
-		super( ...arguments );
-		this.state = {
-			opened: props.initialOpen === undefined ? true : props.initialOpen,
-		};
-		this.toggle = this.toggle.bind( this );
+export function PanelBody(
```

Comment:
This has an open PR here for the refactor #23065 is this based on it?

Reply:
Oh! Not at all. I didn't know that https://github.com/WordPress/gutenberg/pull/23065/files existed. I refactored in order to integrate with Reakit